cf/data/data.py

```diff
@@ -10957,7 +10957,8 @@ def flip(self, axes=None, inplace=False, i=False):
             return d
 
         index = [
-            slice(None, None, -1) if i in axes else slice(None) for i in iaxes
+            slice(None, None, -1) if i in iaxes else slice(None)
+            for i in range(len(d._axes))
```

Comment:
@davidhassell note that the use of `range(len(d._axes))` or some equivalent expression is the key fix here, I think.

Reply:
Hi Sadie - Thanks for spotting and fixing this. It might be better to use `range(d.ndim)` rather than rely on the `_axes` parameter. Do you want to put the fix in?